python-package/securexgboost/remote_server.py

```diff
@@ -0,0 +1,346 @@
+
+    def rpc_XGBoosterDumpModelEx(self, request, context):
+        """
+        Get encrypted model dump
+        """
+        try:
+            length, sarr = self._synchronize(server.XGBoosterDumpModelEx, request)
+            return remote_pb2.Dump(sarr=sarr, length=length, status=0)
+
+        except Exception as e:
+            e = sys.exc_info()
+            print("Error type: " + str(e[0]))
+            print("Error value: " + str(e[1]))
+            traceback.print_tb(e[2])
+
+            return remote_pb2.Dump(sarr=None, length=None, status=-1)
+
+    def rpc_XGBoosterDumpModelExWithFeatures(self, request, context):
+        """
+        Get encrypted model dump with features
+        """
+        try:
+            length, sarr = self._synchronize(server.XGBoosterDumpModelExWithFeatures, request)
+            return remote_pb2.Dump(sarr=sarr, length=length, status=0)
+
+        except Exception as e:
+            e = sys.exc_info()
+            print("Error type: " + str(e[0]))
+            print("Error value: " + str(e[1]))
+            traceback.print_tb(e[2])
+
+            return remote_pb2.Dump(sarr=None, length=None, status=-1)
+
+    def rpc_XGBoosterGetModelRaw(self, request, context):
+        """
+        Get encrypted raw model dump
+        """
+        try:
+            length, sarr = self._synchronize(server.XGBoosterGetModelRaw, request)
+            return remote_pb2.Dump(sarr=sarr, length=length, status=0)
+
+        except Exception as e:
+            e = sys.exc_info()
+            print("Error type: " + str(e[0]))
+            print("Error value: " + str(e[1]))
+            traceback.print_tb(e[2])
+
+            return remote_pb2.Dump(sarr=None, length=None, status=-1)
+
+    def rpc_XGDMatrixNumCol(self, request, context):
+        """
+        Get number of columns in DMatrix
+        """
+        try:
+            ret = self._synchronize(server.XGDMatrixNumCol, request)
+            return remote_pb2.Integer(value=ret)
+
+        except Exception as e:
+            e = sys.exc_info()
+            print("Error type: " + str(e[0]))
+            print("Error value: " + str(e[1]))
+            traceback.print_tb(e[2])
+
+            return remote_pb2.Integer(value=None)
+
+    def rpc_XGDMatrixNumRow(self, request, context):
+        """
+        Get number of rows in DMatrix
+        """
+        try:
+            ret = self._synchronize(server.XGDMatrixNumRow, request)
+            return remote_pb2.Integer(value=ret)
+
+        except Exception as e:
+            e = sys.exc_info()
+            print("Error type: " + str(e[0]))
+            print("Error value: " + str(e[1]))
+            traceback.print_tb(e[2])
+
+            return remote_pb2.Integer(value=None)
+
+def serve(enclave, num_workers=10, all_users=[]):
+    condition = threading.Condition()
+    command = Command()
+    globals()["all_users"] = all_users
+
+    server = grpc.server(futures.ThreadPoolExecutor(max_workers=num_workers))
```

Comment:
This variable name is the same as the name given to RemoteAPI when it's imported

Reply:
Fixed.